Add hexadecimal literal support for decimal type

pranavan15 opened the pull request:
## Purpose
This PR adds support for hexadecimal floating-point literal assignment to decimal type.
e.g: 
```ballerina
 decimal d = 0x01.1;
```
In addition to this it also adds support for assigning a literal with + sign to decimal type.

* Fixes https://github.com/ballerina-platform/ballerina-lang/issues/14255
* Fixes https://github.com/ballerina-platform/ballerina-lang/issues/14398

MaryamZi commented:
Shall we also update (or at least create an issue for) the argument parser for main function parameters, similar to [what we have done for ints](https://github.com/ballerina-platform/ballerina-lang/blob/master/bvm/ballerina-core/src/main/java/org/ballerinalang/util/cli/ArgumentParser.java#L287)?

pranavan15 commented:
> Shall we also update (or at least create an issue for) the argument parser for main function parameters, similar to [what we have done for ints](https://github.com/ballerina-platform/ballerina-lang/blob/master/bvm/ballerina-core/src/main/java/org/ballerinalang/util/cli/ArgumentParser.java#L287)?

+1. Updated the `ArgumentParser` class to support hex decimal arguments. Also, added test cases to check passing decimal arguments to the main function.

hasithaa commented on bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BDecimal.java:
```diff
@@ -58,11 +59,80 @@ public BDecimal(BigDecimal value) {
+     * @param value Hexadecimal string value that needs to be converted.
+     * @return BigDecimal corresponds to the hexadecimal number provided.
+     */
+    private BigDecimal hexToDecimalFloatingPointNumber(String value) {
```
This should be a static method since you are calling from the constructor.

pranavan15 replied:
Commit b47d555 addresses this.
